src/showcase/app/shared/component-viewer/example-viewer/example-viewer.component.ts

```diff
@@ -1,38 +1,72 @@
 import { Component, Injector, Input, OnInit, ViewContainerRef } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { ExampleData, loadExample } from '@sbb-esta/components-examples';
-import { Observable } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { map, switchMap } from 'rxjs/operators';
 
 import { HtmlLoader } from '../../html-loader.service';
 import { moduleParams } from '../../module-params';
 
+interface ExampleCode {
+  label: string;
+  code: string;
+}
+
 @Component({
   selector: 'sbb-example-viewer',
   templateUrl: './example-viewer.component.html',
   styleUrls: ['./example-viewer.component.css'],
 })
 export class ExampleViewerComponent implements OnInit {
   @Input() exampleData: ExampleData;
-  html: Observable<string>;
-  ts: Observable<string>;
-  css: Observable<string>;
+  exampleCodes: Observable<ExampleCode[]>;
   showSource = false;
-
+  private _exampleFileOrder = ['html', 'ts', 'css'];
   constructor(private _htmlLoader: HtmlLoader, private _route: ActivatedRoute) {}
 
-  ngOnInit(): void {
-    const exampleName = this.exampleData.selectorName.replace('sbb-', '').replace('-example', '');
+  // Returns the path to the html file for a given example file.
+  private static _convertToHtmlFilePath(filePath: string): string {
+    return filePath.replace(/(.*)[.](html|ts|css)/, '$1-$2.html');
+  }
+
+  // Get the example name from the selector name
+  private static _convertToExampleName(selectorName: string): string {
+    return selectorName.replace('sbb-', '').replace('-example', '');
+  }
 
-    this.html = this._createLoader(exampleName, 'html');
-    this.ts = this._createLoader(exampleName, 'ts');
-    this.css = this._createLoader(exampleName, 'css');
+  // Get the extension for a given file path
+  private static _getFileExtension(filePath: string): string {
+    return filePath.split('.').pop();
```

Comment:
If we have additional files in the future (e.g. for lightbox or dialog) this would still be reduced to .ts or .html. Could we maybe extract the extension from the "default" files and leave it for the rest?

Reply:
Sorry for missing the lightbox and dialog examples. I've added a filter to the `exampleCodes` observable to ensure that only files whose name match the `indexFileName` (without extension) are displayed.

Reply:
I filtered out other example files than the default ones. However, after reading your comment again, I am not sure if I understood you right. Did you want to filter them out or instead display the file names of the non-default tiles.

E.g. titles for lightbox: TS, HTML, CSS, lightbox-example-content.ts